server/schemas/offlinePayment.schema.js

```diff
@@ -9,12 +9,33 @@ import {
   optionalObjectId,
   nonRequiredString,
 } from './helper.schema';
+import { VALID_TRANSACTION_MODELS } from '../helpers/constants';
+
+const modelType = Joi.string()
+  .label('Model Type')
+  .valid(...Object.values(VALID_TRANSACTION_MODELS).map(String))
+  .required();
+
+const optionalModelType = Joi.string()
+  .label('Model Type')
+  .valid(...Object.values(VALID_TRANSACTION_MODELS).map(String))
+  .optional();
+
+const model = Joi.object().keys({
+  id: requiredObjectId('Model id'),
+  type: modelType,
+});
+
+const optionalModel = Joi.object().keys({
+  id: optionalObjectId('Model Id'),
+  type: optionalModelType,
+});
 
 export const addOfflinePaymentSchema = Joi.object({
   amount: requiredNumber('Amount'),
   bankId: requiredObjectId('Bank id'),
   dateOfPayment: requiredDate('Payment Date'),
-  offerId: requiredObjectId('Offer id'),
+  model,
```

Comment:
Sometimes, this implementation usually feels difficult to read but I usually overlook them. I believe it would be easier to read if you name the validation with `requiredModel` instead of model.

```suggestion
  model: requiredModel,
```